datafusion/physical-expr/src/expressions/binary/kernels.rs

```diff
@@ -24,7 +24,7 @@ use arrow::compute::kernels::bitwise::{
     bitwise_xor, bitwise_xor_scalar,
 };
 use arrow::datatypes::DataType;
-use datafusion_common::internal_err;
+use datafusion_common::plan_err;
```

Comment:
Perhaps `not_impl_err` would be more appropriate?

https://docs.rs/datafusion/latest/datafusion/common/macro.not_impl_err.html

Reply:
it seems unlikely that we would support such an operation so `plan_err` should be good right ? However, I may be overlooking some factors. Could you please clarify if there's a specific rationale behind this suggestion that I might have missed?

Reply:
I was thinking that most `plan_err` s are issues with a user's query (some bug they should fix) where as `not_impl` signals that there isn't some bug with the query

This is kind of documented here: 
* https://docs.rs/datafusion/latest/datafusion/common/enum.DataFusionError.html#variant.Plan
* https://docs.rs/datafusion/latest/datafusion/common/enum.DataFusionError.html#variant.NotImplemented

Reply:
Thats clear. Thank You